src/app/actions.ts

```diff
@@ -19,23 +17,38 @@ export type FlattenedRegion = {
   fullHierarchyName: string;
 };
 
-const fuseOptions = {
+export async function getRegion(regionCode: string) {
+  const regions = flattenedRegionCodes as Region[];
+
+  const region = regions.find((r) => r.code === regionCode);
+  if (!region) {
+    throw new Error(`Could not find region with code ${regionCode}`);
+  }
+  return region;
+}
```

Comment:
This is the _primary_ change here. It looks over the regions data and finds the Region with the matching region code.